SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -118,200 +123,72 @@ def evaluate_integral(self, receiver_location, components, tolerance=1e-4):
-                            )
-                        )
-
-                    if "gxz" in components:
-                        rows["gxz"] -= (
-                            (-1) ** aa
-                            * (-1) ** bb
-                            * (-1) ** cc
-                            * (
-                                np.log(dy_r)
-                                + dz[:, cc] ** 2.0 / (r * dy_r)
-                                + dy[:, bb] / r
-                                - 1.0
-                                / (1 + arg ** 2.0)
-                                * (dy[:, bb] / (r ** 2))
-                                * (r - dz[:, cc] ** 2.0 / r)
-                            )
-                        )
-
-                    arg = dx[:, aa] * dz[:, cc] / dyr
-
-                    if (
-                        ("gyy" in components)
-                        or ("gzz" in components)
-                        or ("guv" in components)
-                    ):
-                        gyy -= (
-                            (-1) ** aa
-                            * (-1) ** bb
-                            * (-1) ** cc
-                            * (
-                                dxdy / (r * dz_r)
-                                + dydz / (r * dx_r)
-                                - np.arctan(arg)
-                                + dy[:, bb]
-                                * (1.0 / (1 + arg ** 2.0))
-                                * dxdz
-                                / dyr ** 2.0
-                                * (r + dy[:, bb] ** 2.0 / r)
-                            )
-                        )
-
-                    if "gyz" in components:
-                        rows["gyz"] -= (
-                            (-1) ** aa
-                            * (-1) ** bb
-                            * (-1) ** cc
-                            * (
-                                np.log(dx_r)
-                                + dz[:, cc] ** 2.0 / (r * (dx_r))
-                                + dx[:, aa] / r
-                                - 1.0
-                                / (1 + arg ** 2.0)
-                                * (dx[:, aa] / (r ** 2))
-                                * (r - dz[:, cc] ** 2.0 / r)
-                            )
-                        )
-
-        if "gyy" in components:
-            rows["gyy"] = gyy
-
-        if "gxx" in components:
-            rows["gxx"] = gxx
-
+        dr = self._nodes - receiver_location
+        dx = dr[..., 0]
+        dy = dr[..., 1]
+        dz = dr[..., 2]
+
+        node_evals = {}
+        if "gx" in components:
+            node_evals["gx"] = prism_fz(dy, dz, dx)
+        if "gy" in components:
+            node_evals["gy"] = prism_fz(dz, dx, dy)
+        if "gz" in components:
+            node_evals["gz"] = prism_fz(dx, dy, dz)
+        if "gxy" in components:
+            node_evals["gxy"] = prism_fzx(dy, dz, dx)
```

Comment:
I am sure you have a smart reason - but why are we getting the `xy` component from the `fzx` call?

Reply:
you can cycle the inputs to the function to get different components

Reply:
very cool